accepted/00000-modular-repos-with-clm.md

```diff
@@ -0,0 +1,194 @@
+> software management tools to handle module bundles.  This section is
+> populated by the module build system.
+
+*modulemd specification*
+
+This situation is handled with the following additional procedure:
+
+Instead of a plain list of RPMs, the API returns a map of package names provided in the `api` section to non-empty lists of artifacts that match the corresponding package names. If a module lists a name in the `api` section but does not provide any artifacts, that name will not be included in the map.
+
+Since not all of the artifacts are listed in the `api` section, the remaining artifacts are put into a special "null" key.
+
+For each key of the map, the backend gets the packages with the same name from all available sources. If any of the packages are not included in the corresponding artifacts list, the package is added to the blacklist.
+
+As a result, when a module defines a package name in its API *and* provides an actual RPM with it, all the other RPMs with the same name are effectively blacklisted even if they are in a different source.
+
+### Module updates and multiple versions
+Every released update of a module is added into the module metadata as a separate module entry with a new `version` number. By design, Uyuni needs to include either all versions of a module or none at all.
+This can be achieved by enabling all versions of the module in the API at the same time. As a result, all the artifacts from all versions will be included sharing the same key in the returning map.
+
+### Error reporting & feedback interface
+To provide a convenient user experience, we must provide feedback for user's actions as early as possible. For this purpose, the CLM project page will have a new panel to display feedback in the form of a list of info, warning, alert items. Each item will provide feedback for a different aspect of the current project setup. That means, each project entity (a filter, a source, etc.) will report its own feedback in case it has one. This will be implemented in a generic way which can be extended to the existing features, by adding a separate data item to the `ProjectResponse` class. Doing so will make it possible for each entity to report its own messages whenever a project response is rendered.
+
+The functionality will be implemented through a new interface called `ProjectEntity`, which will declare a method to list any feedback messages that the entity currently has. An entity can inspect the `ContentProject` object and determine if it has any feedback messages and return them with this method. The method will be called for every entity during the rendering of the project response.
+
+In contrast to using flash messages, the feedback messages generated by the project entities will persist in the UI as long as the project's state is unchanged, making sure the messages get the user's attention and encouraging the user to modify the project to make any warnings/errors disappear.
```

Comment:
:+1: I think the right place to present these messages is right before the Build button and after the Filter box. In case of any `[ERROR]` is present, we can even decide to disable the Build as well.